FixedPositionContainer fails to compile on QNX QCC

## Required information
I just pulled the latest master and after building a project using Iceoryx for QNX, I encounter this error:

```
In file included from external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/port_introspection.hpp:24,
                 from external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.hpp:36,
                 from external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/mempool_introspection.hpp:22,
                 from external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:24,
                 from external/iceoryx/iceoryx_posh/source/roudi/application/iceoryx_roudi_app.cpp:21:
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp: In instantiation of 'class iox::FixedPositionContainer<iox::popo::SubscriberPortData, 1024>::IteratorBase<(iox::FixedPositionContainer<iox::popo::SubscriberPortData, 1024>::IterMutability)0>':
external/iceoryx/iceoryx_posh/include/iceoryx_posh/roudi/port_pool.inl:38:49:   required from here
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:46:15: error: template parameter 'iox::FixedPositionContainer<T, CAPACITY>::IterMutability <anonymous>'
     template <IterMutability>
               ^~~~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:198:22: error: redeclared here as 'iox::FixedPositionContainer<iox::popo::SubscriberPortData, 1024>::IterMutability <anonymous>'
         friend class IteratorBase;
                      ^~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp: In instantiation of 'class iox::FixedPositionContainer<iox::popo::PublisherPortData, 512>::IteratorBase<(iox::FixedPositionContainer<iox::popo::PublisherPortData, 512>::IterMutability)0>':
external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.inl:33:39:   required from here
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:46:15: error: template parameter 'iox::FixedPositionContainer<T, CAPACITY>::IterMutability <anonymous>'
     template <IterMutability>
               ^~~~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:198:22: error: redeclared here as 'iox::FixedPositionContainer<iox::popo::PublisherPortData, 512>::IterMutability <anonymous>'
         friend class IteratorBase;
```

This most likely relates to #2044 

Also note - everything compiles and runs fine on Linux GCC X86/Arm. 

**Operating system:**
QNX 7.10 virtual machine. 

However the compilation happens through some licensed worker somewhere (Bazel magic). It compiles for X86_64

**Compiler version:**
`qcc -V8.3.0,gcc_ntox86_64_cxx`

**Eclipse iceoryx version:**
Current master as of time of writing, more or less

**Observed result or behaviour:**
Libraries that depend on iceoryx fail to compile. I haven't narrowed it down too much which specific iceoryx targets fail - at the very least `iceoryx_dust` as indicated by the error trace provided above.

**Expected result or behaviour:**
Iceoryx continues to compile on qcc

**Conditions where it occurred / Performed steps:**
Compile something that depends transitively on the FixedPositionContainer, with qcc.

Now I see
```
In file included from external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/port_introspection.hpp:24,
                 from external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.hpp:36,
                 from external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/process_manager.hpp:23,
                 from external/iceoryx/iceoryx_posh/source/roudi/process_manager.cpp:19:
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp: In instantiation of 'class iox::FixedPositionContainer<iox::popo::SubscriberPortData, 1024>::IteratorBase<(iox::FixedPositionContainer<iox::popo::SubscriberPortData, 1024>::IterMutability)0>':
external/iceoryx/iceoryx_posh/include/iceoryx_posh/roudi/port_pool.inl:38:49:   required from here
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:55:30: error: template parameter 'iox::FixedPositionContainer<T, CAPACITY>::IterMutability ITER_MUTABILITY'
     template <IterMutability ITER_MUTABILITY>
                              ^~~~~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:67:22: error: redeclared here as 'iox::FixedPositionContainer<iox::popo::SubscriberPortData, 1024>::IterMutability <anonymous>'
         friend class IteratorBase;
                      ^~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp: In instantiation of 'class iox::FixedPositionContainer<iox::popo::PublisherPortData, 512>::IteratorBase<(iox::FixedPositionContainer<iox::popo::PublisherPortData, 512>::IterMutability)0>':
external/iceoryx/iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.inl:33:39:   required from here
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:55:30: error: template parameter 'iox::FixedPositionContainer<T, CAPACITY>::IterMutability ITER_MUTABILITY'
     template <IterMutability ITER_MUTABILITY>
                              ^~~~~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:67:22: error: redeclared here as 'iox::FixedPositionContainer<iox::popo::PublisherPortData, 512>::IterMutability <anonymous>'
         friend class IteratorBase;
                      ^~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp: In instantiation of 'class iox::FixedPositionContainer<iox::runtime::Heartbeat, 300>::IteratorBase<(iox::FixedPositionContainer<iox::runtime::Heartbeat, 300>::IterMutability)0>':
external/iceoryx/iceoryx_posh/source/roudi/process_manager.cpp:298:51:   required from here
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:55:30: error: template parameter 'iox::FixedPositionContainer<T, CAPACITY>::IterMutability ITER_MUTABILITY'
     template <IterMutability ITER_MUTABILITY>
                              ^~~~~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:67:22: error: redeclared here as 'iox::FixedPositionContainer<iox::runtime::Heartbeat, 300>::IterMutability <anonymous>'
         friend class IteratorBase;
                      ^~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp: In instantiation of 'class iox::FixedPositionContainer<iox::runtime::Heartbeat, 300>::IteratorBase<(iox::FixedPositionContainer<iox::runtime::Heartbeat, 300>::IterMutability)1>':
external/iceoryx/iceoryx_posh/source/roudi/process_manager.cpp:370:49:   required from here
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:55:30: error: template parameter 'iox::FixedPositionContainer<T, CAPACITY>::IterMutability ITER_MUTABILITY'
     template <IterMutability ITER_MUTABILITY>
                              ^~~~~~~~~~~~~~~
external/iceoryx/iceoryx_dust/container/include/iox/fixed_position_container.hpp:67:22: error: redeclared here as 'iox::FixedPositionContainer<iox::runtime::Heartbeat, 300>::IterMutability <anonymous>'
         friend class IteratorBase;
```

--- a/.cirrus.yaml
+++ b/.cirrus.yaml
@@ -31,23 +31,39 @@ iox_common_cpu_and_memory_config_for_tests_template: &IOX_COMMON_CPU_AND_MEMORY_
 
 # Build and Test
 
-iox_posix_clean_build_script_template: &IOX_POSIX_CLEAN_BUILD
+iox_posix_pre_cmake_config_script_template: &IOX_POSIX_PRE_CMAKE_CONFIG
   - echo "#### Deleting build and iox-tests-bin dir"
   - if [[ -d "build" ]]; then rm -rf build; fi
   - if [[ -d "iox-tests-bin" ]]; then rm -rf iox-tests-vin; fi
   - echo "#### Running cmake to configure build"
-  - cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=ON -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DCMAKE_INSTALL_PREFIX=build/install
+
+iox_posix_post_cmake_config_script_template: &IOX_POSIX_POST_CMAKE_CONFIG
   - echo "#### Running cmake to build with CIRRUS_CPU = " $CIRRUS_CPU
   - cmake --build build --target install -j $CIRRUS_CPU
 
-iox_posix_clean_build_with_additional_user_template: &IOX_POSIX_CLEAN_BUILD_WITH_ADDITIONAL_USER
-  - echo "#### Deleting build and iox-tests-bin dir"
-  - if [[ -d "build" ]]; then rm -rf build; fi
-  - if [[ -d "iox-tests-bin" ]]; then rm -rf iox-tests-vin; fi
-  - echo "#### Running cmake to configure build"
+iox_posix_cmake_config_strict_with_no_additional_user_script_template: &IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_NO_ADDITIONAL_USER
+  - cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=ON -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=OFF -DCMAKE_INSTALL_PREFIX=build/install
+
+iox_posix_cmake_config_strict_with_additional_user_script_template: &IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_ADDITIONAL_USER
   - cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=ON -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=ON -DCMAKE_INSTALL_PREFIX=build/install
-  - echo "#### Running cmake to build with CIRRUS_CPU = " $CIRRUS_CPU
-  - cmake --build build --target install -j $CIRRUS_CPU
+
+iox_posix_cmake_config_no_strict_with_additional_user_script_template: &IOX_POSIX_CMAKE_CONFIG_NO_STRICT_WITH_ADDITIONAL_USER
+  - cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=OFF -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=ON -DCMAKE_INSTALL_PREFIX=build/install
+
+iox_posix_clean_build_strict_with_no_additional_user_script_template: &IOX_POSIX_CLEAN_BUILD_STRICT_WITH_NO_ADDITIONAL_USER
+  <<: *IOX_POSIX_PRE_CMAKE_CONFIG
+  <<: *IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_NO_ADDITIONAL_USER
+  <<: *IOX_POSIX_POST_CMAKE_CONFIG
+
+iox_posix_clean_build_strict_with_additional_user_template: &IOX_POSIX_CLEAN_BUILD_STRICT_WITH_ADDITIONAL_USER
+  <<: *IOX_POSIX_PRE_CMAKE_CONFIG
+  <<: *IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_ADDITIONAL_USER
+  <<: *IOX_POSIX_POST_CMAKE_CONFIG
+
+iox_posix_clean_build_no_strict_with_additional_user_allow_warnings_template: &IOX_POSIX_CLEAN_BUILD_NO_STRICT_WITH_ADDITIONAL_USER_ALLOW_WARNING
+  <<: *IOX_POSIX_PRE_CMAKE_CONFIG
+  <<: *IOX_POSIX_CMAKE_CONFIG_NO_STRICT_WITH_ADDITIONAL_USER
+  <<: *IOX_POSIX_POST_CMAKE_CONFIG
 
 iox_prepare_test_binaries_for_cache_template: &IOX_PREPARE_TEST_BINARIES_FOR_CACHE
   - mkdir -p iox-tests-bin
@@ -101,7 +117,7 @@ ubuntu_22_04_aarch64_build_task:
     reupload_on_changes: true
     fingerprint_key: $CIRRUS_OS_ubuntu_22_04_aarch64_test_binaries_cache
   build_script:
-    <<: *IOX_POSIX_CLEAN_BUILD_WITH_ADDITIONAL_USER
+    <<: *IOX_POSIX_CLEAN_BUILD_STRICT_WITH_ADDITIONAL_USER
   populate_test_binary_folder_script:
     <<: *IOX_PREPARE_TEST_BINARIES_FOR_CACHE
 
@@ -115,6 +131,42 @@ ubuntu_22_04_aarch64_test_task:
     folder: iox-tests-bin
     reupload_on_changes: false
     fingerprint_key: $CIRRUS_OS_ubuntu_22_04_aarch64_test_binaries_cache
+  test_script:
+    <<: *IOX_RUN_TESTS
+
+#
+# Arch Linux x64 gcc 8.3 aka QNX canary build
+#
+
+arch_linux_x64_gcc_8_3_aka_qnx_canary_build_task:
+  depends_on: preflight_check
+  container:
+    dockerfile: tools/ci/docker/archlinux-base-devel
+    <<: *IOX_COMMON_CPU_AND_MEMORY_CONFIG_FOR_BUILDS
+  <<: *IOX_TASK_TIMEOUT
+  test_binaries_cache:
+    folder: iox-tests-bin
+    reupload_on_changes: true
+    fingerprint_key: $CIRRUS_OS_archlinux_x64_gcc_8_3_aka_qnx_canary_test_binaries_cache
+  env:
+    # use GCC 8.3 which corresponds to QCC 8.3 on QNX 7.1
+    CC: gcc-8
+    CXX: g++-8
+  build_script:
+    <<: *IOX_POSIX_CLEAN_BUILD_NO_STRICT_WITH_ADDITIONAL_USER_ALLOW_WARNING
+  populate_test_binary_folder_script:
+    <<: *IOX_PREPARE_TEST_BINARIES_FOR_CACHE
+
+arch_linux_x64_gcc_8_3_aka_qnx_canary_test_task:
+  depends_on: arch_linux_x64_gcc_8_3_aka_qnx_canary_build
+  container:
+    dockerfile: tools/ci/docker/archlinux-base-devel
+    <<: *IOX_COMMON_CPU_AND_MEMORY_CONFIG_FOR_TESTS
+  <<: *IOX_TASK_TIMEOUT
+  test_binaries_cache:
+    folder: iox-tests-bin
+    reupload_on_changes: false
+    fingerprint_key: $CIRRUS_OS_archlinux_x64_gcc_8_3_aka_qnx_canary_test_binaries_cache
   test_script:
     <<: *IOX_RUN_TESTS
 
@@ -133,7 +185,7 @@ arch_linux_x64_build_task:
     reupload_on_changes: true
     fingerprint_key: $CIRRUS_OS_archlinux_x64_test_binaries_cache
   build_script:
-    <<: *IOX_POSIX_CLEAN_BUILD_WITH_ADDITIONAL_USER
+    <<: *IOX_POSIX_CLEAN_BUILD_STRICT_WITH_ADDITIONAL_USER
   populate_test_binary_folder_script:
     <<: *IOX_PREPARE_TEST_BINARIES_FOR_CACHE
 
@@ -168,7 +220,7 @@ freebsd_x64_build_task:
     - pkg install -y cmake git ncurses bash wget
     - ln -s /usr/local/bin/bash /bin/bash
   build_script:
-    <<: *IOX_POSIX_CLEAN_BUILD
+    <<: *IOX_POSIX_CLEAN_BUILD_STRICT_WITH_NO_ADDITIONAL_USER
   populate_test_binary_folder_script:
     <<: *IOX_PREPARE_TEST_BINARIES_FOR_CACHE
 
@@ -202,7 +254,7 @@ macos_aarch64_build_task:
   setup_script:
     - brew install ncurses
   build_script:
-    <<: *IOX_POSIX_CLEAN_BUILD
+    <<: *IOX_POSIX_CLEAN_BUILD_STRICT_WITH_NO_ADDITIONAL_USER
   populate_test_binary_folder_script:
     <<: *IOX_PREPARE_TEST_BINARIES_FOR_CACHE
 

--- a/.cirrus.yaml
+++ b/.cirrus.yaml
@@ -44,38 +44,40 @@ iox_common_cpu_and_memory_config_for_tests_template: &IOX_COMMON_CPU_AND_MEMORY_
 # Build and Test
 
 iox_posix_pre_cmake_config_script_template: &IOX_POSIX_PRE_CMAKE_CONFIG
-  - echo "#### Deleting build and iox-tests-bin dir"
-  - if [[ -d "build" ]]; then rm -rf build; fi
-  - if [[ -d "iox-tests-bin" ]]; then rm -rf iox-tests-vin; fi
-  - echo "#### Running cmake to configure build"
+  echo "#### Deleting build and iox-tests-bin dir"
+  && if [[ -d "build" ]]; then rm -rf build; fi
+  && if [[ -d "iox-tests-bin" ]]; then rm -rf iox-tests-vin; fi
 
 iox_posix_post_cmake_config_script_template: &IOX_POSIX_POST_CMAKE_CONFIG
-  - echo "#### Running cmake to build with CIRRUS_CPU = " $CIRRUS_CPU
-  - cmake --build build --target install -j $CIRRUS_CPU
+  echo "#### Running cmake to build with CIRRUS_CPU = " $CIRRUS_CPU
+  && cmake --build build --target install -j $CIRRUS_CPU
 
 iox_posix_cmake_config_strict_with_no_additional_user_script_template: &IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_NO_ADDITIONAL_USER
-  - cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=ON -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=OFF -DCMAKE_INSTALL_PREFIX=build/install
+  echo "#### Running cmake to configure build"
+  && cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=ON -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=OFF -DCMAKE_INSTALL_PREFIX=build/install
 
 iox_posix_cmake_config_strict_with_additional_user_script_template: &IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_ADDITIONAL_USER
-  - cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=ON -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=ON -DCMAKE_INSTALL_PREFIX=build/install
+  echo "#### Running cmake to configure build"
+  && cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=ON -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=ON -DCMAKE_INSTALL_PREFIX=build/install
 
 iox_posix_cmake_config_no_strict_with_additional_user_script_template: &IOX_POSIX_CMAKE_CONFIG_NO_STRICT_WITH_ADDITIONAL_USER
-  - cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=OFF -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=ON -DCMAKE_INSTALL_PREFIX=build/install
+  echo "#### Running cmake to configure build"
+  && cmake -Bbuild -Hiceoryx_meta -DBUILD_STRICT=OFF -DBINDING_C=ON -DINTROSPECTION=ON -DEXAMPLES=ON -DBUILD_TEST=ON -DTEST_ADD_USER=ON -DCMAKE_INSTALL_PREFIX=build/install
 
-iox_posix_clean_build_strict_with_no_additional_user_script_template: &IOX_POSIX_CLEAN_BUILD_STRICT_WITH_NO_ADDITIONAL_USER
-  <<: *IOX_POSIX_PRE_CMAKE_CONFIG
-  <<: *IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_NO_ADDITIONAL_USER
-  <<: *IOX_POSIX_POST_CMAKE_CONFIG
+iox_posix_clean_build_strict_with_no_additional_user_template: &IOX_POSIX_CLEAN_BUILD_STRICT_WITH_NO_ADDITIONAL_USER
+  - *IOX_POSIX_PRE_CMAKE_CONFIG
+  - *IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_NO_ADDITIONAL_USER
+  - *IOX_POSIX_POST_CMAKE_CONFIG
 
 iox_posix_clean_build_strict_with_additional_user_template: &IOX_POSIX_CLEAN_BUILD_STRICT_WITH_ADDITIONAL_USER
-  <<: *IOX_POSIX_PRE_CMAKE_CONFIG
-  <<: *IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_ADDITIONAL_USER
-  <<: *IOX_POSIX_POST_CMAKE_CONFIG
+  - *IOX_POSIX_PRE_CMAKE_CONFIG
+  - *IOX_POSIX_CMAKE_CONFIG_STRICT_WITH_ADDITIONAL_USER
+  - *IOX_POSIX_POST_CMAKE_CONFIG
 
-iox_posix_clean_build_no_strict_with_additional_user_allow_warnings_template: &IOX_POSIX_CLEAN_BUILD_NO_STRICT_WITH_ADDITIONAL_USER_ALLOW_WARNING
-  <<: *IOX_POSIX_PRE_CMAKE_CONFIG
-  <<: *IOX_POSIX_CMAKE_CONFIG_NO_STRICT_WITH_ADDITIONAL_USER
-  <<: *IOX_POSIX_POST_CMAKE_CONFIG
+iox_posix_clean_build_no_strict_with_additional_user_template: &IOX_POSIX_CLEAN_BUILD_NO_STRICT_WITH_ADDITIONAL_USER
+  - *IOX_POSIX_PRE_CMAKE_CONFIG
+  - *IOX_POSIX_CMAKE_CONFIG_NO_STRICT_WITH_ADDITIONAL_USER
+  - *IOX_POSIX_POST_CMAKE_CONFIG
 
 iox_prepare_test_binaries_for_cache_template: &IOX_PREPARE_TEST_BINARIES_FOR_CACHE
   - mkdir -p iox-tests-bin
@@ -165,7 +167,7 @@ arch_linux_x64_gcc_8_3_aka_qnx_canary_build_task:
     CC: gcc-8
     CXX: g++-8
   build_script:
-    <<: *IOX_POSIX_CLEAN_BUILD_NO_STRICT_WITH_ADDITIONAL_USER_ALLOW_WARNING
+    <<: *IOX_POSIX_CLEAN_BUILD_STRICT_WITH_ADDITIONAL_USER
   populate_test_binary_folder_script:
     <<: *IOX_PREPARE_TEST_BINARIES_FOR_CACHE
 

--- a/.cirrus.yaml
+++ b/.cirrus.yaml
@@ -13,6 +13,18 @@
 ---
 
 #
+# Filter to run the CI only on the master and release branches or for pull request to the master, release* and iox-* branches
+#
+
+only_if: $CIRRUS_BRANCH == 'master'
+         || $CIRRUS_BRANCH == 'release*'
+         || ($CIRRUS_PR != '' && ($CIRRUS_BASE_BRANCH == 'master'
+                                  || $CIRRUS_BASE_BRANCH == 'release*'
+                                  || $CIRRUS_BASE_BRANCH == 'iox-*'
+                                  )
+            )
+
+#
 # Templates
 #
 

--- a/doc/website/release-notes/iceoryx-unreleased.md
+++ b/doc/website/release-notes/iceoryx-unreleased.md
@@ -101,7 +101,7 @@
 - Fix musl libc compile (missing sys/stat.h include in mqueue.h for mode_t definition) [\#2072](https://github.com/eclipse-iceoryx/iceoryx/issues/2072)
 - cxx::Expects macro conflicts with Microsoft GSL Expects macro [#2080](https://github.com/eclipse-iceoryx/iceoryx/issues/2080)
 - Implement move/copy constructor and assignment for `FixedPositionContainer` [#2052](https://github.com/eclipse-iceoryx/iceoryx/issues/2052)
-
+- FixedPositionContainer fails to compile on QNX QCC [#2084](https://github.com/eclipse-iceoryx/iceoryx/issues/2084)
 
 **Refactoring:**
 

--- a/iceoryx_dust/container/include/iox/fixed_position_container.hpp
+++ b/iceoryx_dust/container/include/iox/fixed_position_container.hpp
@@ -194,9 +194,6 @@ class FixedPositionContainer final
 
         friend class FixedPositionContainer;
 
-        template <IterMutability>
-        friend class IteratorBase;
-
         /// @brief Construct a const iterator from an iterator
         // NOLINTJUSTIFICATION conversion from non const iterator to const iterator follows the STL behavior
         // NOLINTNEXTLINE(hicpp-explicit-conversions)

--- a/tools/ci/docker/archlinux-base-devel
+++ b/tools/ci/docker/archlinux-base-devel
@@ -9,11 +9,15 @@ ARG DEBIAN_FRONTEND=noninteractive
 # Minimize the number of layers (RUN, COPY and ADD create layers)
 # https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers
 # Install additional packages
-RUN pacman -Syu --noconfirm \
-  clang \
-  cmake \
-  git \
-  ncurses \
+RUN \
+  pacman -Syu --noconfirm \
+    clang \
+    cmake \
+    git \
+    ncurses \
+  && pacman -U --noconfirm \
+    https://archive.archlinux.org/packages/g/gcc8-libs/gcc8-libs-8.3.0-4-x86_64.pkg.tar.xz \
+    https://archive.archlinux.org/packages/g/gcc8/gcc8-8.3.0-4-x86_64.pkg.tar.xz \
   && pacman -Scc --noconfirm \
   && useradd iox_roudi_test1 \
   && useradd iox_roudi_test2 \

